auth/validator.go

```diff
@@ -43,57 +42,29 @@ func (m *Middleware) parseAndValidateJWT(rawToken string) (Token, error) {
 }
 
 func (m *Middleware) verifySignature(t Token, keySet *oidcclient.OIDCTenant) (err error) {
-	headers, err := getHeaders(t.GetTokenValue())
+	headers, err := getHeaders(t.TokenValue())
 	if err != nil {
 		return err
 	}
-	kid := headers.KeyID()
 	alg := headers.Algorithm()
 
 	// fail early to avoid another parsing of encoded token
 	if alg == "" {
 		return errors.New("alg is missing from jwt header")
 	}
 
-	publicKey, err := getPublicKey(kid, keySet)
+	// parse and verify signature
+	jwks, err := keySet.GetJWKs()
 	if err != nil {
 		return err
 	}
-
-	// parse and verify signature
-	_, err = jwt.ParseString(t.GetTokenValue(), jwt.WithVerify(alg, publicKey))
+	_, err = jwt.ParseString(t.TokenValue(), jwt.WithKeySet(jwks), jwt.UseDefaultKey(true))
```

Comment:
hi @f-blass i guess useDefaultKey(true) takes first one, in case no kid is provided with token header?

Reply:
If no kid is provided in the token header it will verify the token, if the server only provides one key. (Basically the same logic I had implemented by myself before)
If the server provides multiple keys and no kid in the token, verficication will fail